File: 期末考/Q6 Python繼承/Cuboid.py
from Shape import Shape

class Cuboid(Shape):
    def __init__(self, newLength, newWidth, newHeight, newColor):
        self.__length = newLength
        self.__width = newWidth
        self.__height = newHeight
        super().__init__(newColor)
    def setLength(self, newLength):
        self.__length = newLength
    def getLength(self):
        return self.__length
    def setWidth(self, newWidth):
        self.__width = newWidth
    def getWidth(self):
        return self.__width
    def setHeight(self, newHeight):
        self.__height = newHeight
    def getHeight(self):
        return self.__height
    def getVolume(self):
        return self.getLength() * self.getWidth() * self.getHeight()
    def __str__(self):
        return "Cuboid"

File: 期末考/Q3 C++多型/Cuboid.cpp
#include "Cuboid.h"
Cuboid::Cuboid(double length, double width, double height, string color)
{
    setLength(length);
    setWidth(width);
    setHeight(height);
    setColor(color);
}
void Cuboid::setLength(double newLength)
{
    length = newLength;
}
void Cuboid::setWidth(double newWidth)
{
    width = newWidth;
}
void Cuboid::setHeight(double newHeight)
{
    height = newHeight;
}
double Cuboid::getLength()
{
    return length;
}
double Cuboid::getWidth()
{
    return width;
}
double Cuboid::getHeight()
{
    return height;
}
double Cuboid::getVolume()
{
    return length * width * height;
}
void Cuboid::toString()
{
    cout << "Cuboid:(" << fixed << setprecision(0) << length << "," << width << "," << height << "),";
}
